test-pyproject-toml

hadijannat opened the pull request.

s-heppner commented:
It seems the toml file is not recognized as the `setup.py` was, since the CI is failing at the install phase. 

Does that mean we have to adapt the CI configuration?
Do we need to update our installation instructions in the README?

hadijannat commented:
> It seems the toml file is not recognized as the `setup.py` was, since the CI is failing at the install phase.
> 
> Does that mean we have to adapt the CI configuration? Do we need to update our installation instructions in the README?



> It seems the toml file is not recognized as the `setup.py` was, since the CI is failing at the install phase.
> 
> Does that mean we have to adapt the CI configuration? Do we need to update our installation instructions in the README?

I am trying to update ci configuration as below: 
'''
jobs:
  test:
    runs-on: ubuntu-latest
    strategy:
      matrix:
        python-version: ["3.8", "3.10", "3.12"]
    env:
      COUCHDB_ADMIN_PASSWORD: "yo0Quai3"
      AAS_SPECS_RELEASE_TAG: "V3.0.7"
    services:
      couchdb:
        image: couchdb:3
        ports:
          - 5984:5984
        env:
          COUCHDB_USER: "admin"
          COUCHDB_PASSWORD: ${{ env.COUCHDB_ADMIN_PASSWORD }}

    steps:
    - uses: actions/checkout@v2
    - name: Set up Python ${{ matrix.python-version }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python-version }}
    - name: Install Poetry
      run: |
        pip install poetry
    - name: Install dependencies with Poetry
      run: |
        poetry install
    - name: Collect schema files from aas-specs
      run: |
        mkdir -p ./test/adapter/schema
        curl -sSL -o ./test/adapter/schema/aasJSONSchema.json https://raw.githubusercontent.com/admin-shell-io/aas-specs/${{ env.AAS_SPECS_RELEASE_TAG }}/schemas/json/aas.json
        curl -sSL -o ./test/adapter/schema/aasXMLSchema.xsd https://raw.githubusercontent.com/admin-shell-io/aas-specs/${{ env.AAS_SPECS_RELEASE_TAG }}/schemas/xml/AAS.xsd
    - name: Setup test config and CouchDB database server
      run: |
        python test/_helper/setup_testdb.py -u "admin" -p "$COUCHDB_ADMIN_PASSWORD"
    - name: Test with coverage + unittest
      run: |
        poetry run coverage run --source=basyx -m unittest
    - name: Report test coverage
      if: ${{ always() }}
      run: |
        poetry run coverage report -m
'''

hadijannat committed:
Migrate to pyproject.toml and update related files

s-heppner commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+
+[project]
+name = "basyx-python-sdk"
+version = "0.1.0"  # Replace with your actual version
```
We should replace that with at least the current version 1.0.0

s-heppner commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+version = "0.1.0"  # Replace with your actual version
+description = "The Eclipse BaSyx Python SDK, an implementation of the Asset Administration Shell for Industry 4.0 systems"
+authors = [
+    {name = "The Eclipse BaSyx Authors", email = "example@example.com"}  # Replace with a real email
```
The E-Mail needs replacing, I wonder what we should put there.

hadijannat replied:
Tnx. Which email should I write here?

s-heppner commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+    "Operating System :: OS Independent",
+    "Development Status :: 5 - Production/Stable",
+]
+requires-python = ">=3.6"
```
We do not support Python 3.6 anymore. 
This should be `>=3.8`.

s-heppner commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+]
+requires-python = ">=3.6"
+dependencies = [
+    "python-dateutil>=2.8,<3",
+    "lxml>=4.2,<5",
+    "urllib3>=1.26,<2.0",
+    "pyecma376-2>=0.2.4",
```
Are we sure these are all the dependencies?
In the `requirements.txt` ([here](https://github.com/eclipse-basyx/basyx-python-sdk/blob/main/requirements.txt)), we have:

```
jsonschema~=4.7
lxml>=4.2,<5
python-dateutil>=2.8,<3.0
types-python-dateutil
pyecma376-2>=0.2.4
urllib3>=1.26,<2.0
lxml-stubs~=0.5.1
```

s-heppner commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+[project.urls]
+Homepage = "https://github.com/eclipse-basyx/basyx-python-sdk"
+
+[project.scripts]
+aas-compliance-check = "basyx.aas.compliance_tool.cli:main"
```
What does that line do?
At the moment, there is no such module `basyx.aas.compliance_tool`, since we moved it to https://github.com/rwth-iat/aas-compliance-tool
Could that be an artefact that we accidentally left in the old `setup.py`?

s-heppner commented on pyproject.toml:
```diff
@@ -0,0 +1,39 @@
+
+[tool.setuptools.package-data]
+"basyx" = ["py.typed"]
+"basyx.aas.examples.data" = ["TestFile.pdf"]
```
Seems like there's a new line missing at the end of the file

s-heppner commented on requirements.txt:
```diff
@@ -5,3 +5,7 @@ types-python-dateutil
 pyecma376-2>=0.2.4
 urllib3>=1.26,<2.0
 lxml-stubs~=0.5.1
+
+jsonschema~=4.7
+types-python-dateutil
+lxml-stubs~=0.5.1
```
Why did we change the order of the requirements?
Why does the list differ from the dependencies listed in the `pyproject.toml`?

s-heppner commented on setup.py:
```diff
@@ -1,48 +1,16 @@
     entry_points={
         'console_scripts': [
-            "aas-compliance-check = basyx.aas.compliance_tool.cli:main"
+            "aas-compliance-check=basyx.aas.compliance_tool.cli:main"
         ]
```
Seems like this should be removed here as well, since we moved the compliance tool to a different repository (https://github.com/rwth-iat/aas-compliance-tool)

s-heppner reviewed (changes requested):
Here's my review:

s-heppner commented:
And then furthermore, this PR should probably point at [eclipse-basyx/basyx-python-sdk/main](https://github.com/eclipse-basyx/basyx-python-sdk/tree/main), rather than our fork's main

hadijannat commented:
the PR is already targeting the main branch of the current repository, could you elaborate more about this review: '''And then furthermore, this PR should probably point at [eclipse-basyx/basyx-python-sdk/main](https://github.com/eclipse-basyx/basyx-python-sdk/tree/main), rather than our fork's main'''

hadijannat committed:
Update pyproject.toml and setup.py based on review comments
- Remove reference to aas-compliance-check from pyproject.toml
- Add newline at end of pyproject.toml
- Remove entry point for aas-compliance-check from setup.py

s-heppner commented on pyproject.toml:
```diff
@@ -4,10 +4,10 @@ build-backend = "setuptools.build_meta"
 [project]
 name = "basyx-python-sdk"
-version = "0.1.0"  # Replace with your actual version
+version = "1.0.0"  # actual version
```
The comment is not a particularly good one, as it is obvious that `version` equals the version

s-heppner commented on pyproject.toml:
```diff
@@ -4,10 +4,10 @@ build-backend = "setuptools.build_meta"
 description = "The Eclipse BaSyx Python SDK, an implementation of the Asset Administration Shell for Industry 4.0 systems"
 authors = [
-    {name = "The Eclipse BaSyx Authors", email = "example@example.com"}  # Replace with a real email
+    {name = "The Eclipse BaSyx Authors", email = "h.jannatabadi@iat.rwth-aachen.de"}  # actual authors
```
If you cannot leave out the E-Mail field, I'd use `admins@iat.rwth-aachen.de`.
Same thing as above about the comment.

s-heppner commented on requirements.txt:
```diff
@@ -6,6 +6,3 @@ pyecma376-2>=0.2.4
 
-jsonschema~=4.7
-types-python-dateutil
-lxml-stubs~=0.5.1
```
I'm still not sure about the dependencies, but ignore, if you're not done with them.

s-heppner commented on setup.py:
```diff
@@ -8,9 +8,5 @@
-        ]
-    },
+
 )
```
No new line at the end of the file

hadijannat committed:
Update pyproject.toml, setup.py, and requirements.txt

hadijannat committed:
Update dependency management and CI configuration

hadijannat committed:
Fix MyPy and Sphinx documentation errors

- Add types-python-dateutil to dev dependencies
- Update Duration type handling in datatypes.py
- Add sphinx-argparse to docs dependencies
- Update CI configuration for MyPy and Sphinx

hadijannat committed:
Update docstrings and type annotations for Duration

hadijannat committed:
Update Duration references in docstrings and conf.py

hadijannat committed:
Update Duration references in docstrings and conf.py

hadijannat committed:
Remove requirements.txt and update related files for pyproject.toml migration

hadijannat committed:
Fix: Add newline at end of basyx/aas/__init__.py

hadijannat committed:
Your commit message describing the changes

hadijannat committed:
Clarified Duration alias and updated Sphinx configuration

hadijannat committed:
Updated Duration type definition and Sphinx configuration

hadijannat committed:
Add pyproject.toml and update setup.py for new build system

hadijannat committed:
new pyproject and setup

hadijannat commented:
https://github.com/rwth-iat/basyx-python-sdk/pull/51
